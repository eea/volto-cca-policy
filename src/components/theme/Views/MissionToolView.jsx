import React from 'react';
import {
  HTMLField,
  BannerTitle,
  MetadataItemList,
} from '@eeacms/volto-cca-policy/helpers';
import { Container, Grid, Image, Segment, List } from 'semantic-ui-react';
import {
  PortalMessage,
  MissionDisclaimer,
} from '@eeacms/volto-cca-policy/components';

function MissionToolView(props) {
  const { content } = props;

  return (
    <div className="mission-item-view">
      <BannerTitle
        content={{ ...content, image: '' }}
        data={{
          info: [{ description: '' }],
          hideContentType: true,
          hideCreationDate: false,
          hideModificationDate: false,
          hidePublishingDate: false,
          hideDownloadButton: false,
          hideShareButton: false,
        }}
      />

      <Container>
        <PortalMessage content={content} />
        <Grid columns="12">
          <div className="row">
            <Grid.Column
              mobile={12}
              tablet={12}
              computer={8}
              className="col-left"
            >
              {!!content.objective && (
                <>
                  <h2>Objective(s)</h2>
                  <HTMLField value={content.objective} />
                </>
              )}

              {!!content.short_description && (
                <>
                  <h2>Short description</h2>
                  <HTMLField value={content.short_description} />
                </>
              )}

              {!!content.free_keywords && (
                <>
                  <h2>Free keywords</h2>
                  <HTMLField value={content.free_keywords} />
                </>
              )}

              {!!content.readiness_for_use &&
                content.readiness_for_use.length > 0 && (
                  <>
                    <h2>Readiness for use</h2>

                    <MetadataItemList value={content.readiness_for_use} />
                  </>
                )}

              {!!content.applications && (
                <>
                  <h2>Applications</h2>
                  <HTMLField value={content.applications} />
                </>
              )}

              {!!content.strengths_weaknesses && (
                <>
                  <h2>
                    Strengths and weaknesses, comparative added value to other
                    similar tools
                  </h2>
                  <HTMLField value={content.strengths_weaknesses} />
                </>
              )}

              {!!content.input && (
                <>
                  <h2>Input(s)</h2>
                  <HTMLField value={content.input} />
                </>
              )}

              {!!content.output && (
                <>
                  <h3>Output(s)</h3>
                  <HTMLField value={content.output} />
                </>
              )}

              {!!content.output_image && (
                <Image
                  src={content.output_image?.scales?.large?.download}
                  alt={content.title}
                  style={{ margin: '2.5em 0' }}
                />
              )}

              {!!content.replicability && (
                <>
                  <h2>Replicability: Cost/effort for (new) usage</h2>
                  <HTMLField value={content.replicability} />
                </>
              )}

              {!!content.materials && (
                <>
                  <h2>Materials or other support available</h2>
                  <HTMLField value={content.materials} />
                </>
              )}

              {!!content.website && (
                <>
                  <h2>Website and maintenance</h2>
                  <HTMLField value={content.website} />
                </>
              )}

              {!!content.contact && (
                <>
                  <h2>Contact</h2>
                  <HTMLField value={content.contact} />
                </>
              )}

              {!!content.associated_project && (
                <>
                  <h2>Associated project(s)</h2>
                  <HTMLField value={content.associated_project} />
                </>
              )}
            </Grid.Column>
            <Grid.Column
              mobile={12}
              tablet={12}
              computer={4}
              className="col-right"
            >
              <Segment className="metadata">
                {!!content.rast_steps && content.rast_steps.length > 0 && (
                  <>
                    <h5>RAST step(s) of relevance:</h5>
                    <List>
                      {content.rast_steps.map((step, index) => (
                        <List.Item key={index}>{step.title}</List.Item>
                      ))}
                    </List>
                  </>
                )}

                {!!content.geographical_scale &&
                  content.geographical_scale.length > 0 && (
                    <>
                      <h5>Geographical scale:</h5>
                      <MetadataItemList value={content.geographical_scale} />
                    </>
                  )}

                {!!content.geographical_area && (
                  <>
                    <h5>Geographical area:</h5>
                    <HTMLField value={content.geographical_area} />
                  </>
                )}

                {!!content.climate_impacts &&
                  content.climate_impacts.length > 0 && (
                    <>
                      <h5>Climate impacts:</h5>
                      <MetadataItemList value={content.climate_impacts} />
                    </>
                  )}

                {!!content.tool_language && content.tool_language.length > 0 && (
                  <>
                    <h5>Language(s) of the tool:</h5>
                    <MetadataItemList value={content.tool_language} />
                  </>
                )}

                {!!content.sectors && content.sectors.length > 0 && (
                  <>
                    <h5>Adaptation sectors:</h5>
                    <MetadataItemList value={content.sectors} />
                  </>
                )}

                {!!content.most_useful_for &&
                  content.most_useful_for.length > 0 && (
                    <>
                      <h5>Most useful for:</h5>
                      <MetadataItemList value={content.most_useful_for} />
                    </>
                  )}

                {!!content.user_requirements &&
                  content.user_requirements.length > 0 && (
                    <>
                      <h5>User requirements:</h5>
                      <MetadataItemList value={content.user_requirements} />
                    </>
                  )}
              </Segment>
            </Grid.Column>
          </div>
        </Grid>

        <MissionDisclaimer />
      </Container>
    </div>
  );
}

export default MissionToolView;
