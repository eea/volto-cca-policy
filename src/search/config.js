import { mergeConfig } from '@eeacms/search';

import facets from './facets';

// import views from './views';
// import filters from './filters';
// import vocabs from './vocabulary';

const getClientProxyAddress = () => {
  const url = new URL(window.location);
  url.pathname = '';
  url.search = '';
  return url.toString();
};

const ccaConfig = {
  title: 'ClimateAdapt Main',
};

export default function installMainSearch(config) {
  const envConfig = process.env.RAZZLE_ENV_CONFIG
    ? JSON.parse(process.env.RAZZLE_ENV_CONFIG)
    : ccaConfig;

  const pjson = require('@eeacms/volto-cca-policy/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.ccaSearch = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/globalsearch',
    index_name: 'data_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
  };

  const { ccaSearch } = config.searchui;

  ccaSearch.permanentFilters.push({
    term: {
      cluster_name: 'cca',
    },
  });
  ccaSearch.permanentFilters.push({
    term: {
      cca_include_in_mission: 'true',
    },
  });

  ccaSearch.facets = facets;

  ccaSearch.initialView.tilesLandingPageParams.sections = [
    {
      id: 'types',
      title: 'Types',
      facetField: 'objectProvides',
      sortOn: 'alpha',
      icon: {
        family: 'Content types',
      },
    },
    // {
    //   id: 'topics',
    //   title: 'Topics',
    //   facetField: 'topic',
    //   sortOn: 'alpha',
    //   whitelist: [
    //     'Agriculture and food',
    //     'Air pollution',
    //     'Bathing water quality',
    //     'Biodiversity',
    //     'Bioeconomy',
    //     'Buildings and construction',
    //     'Chemicals',
    //     'Circular economy',
    //     'Climate change adaptation',
    //     'Climate change mitigation',
    //     'Electric vehicles',
    //     'Energy',
    //     'Energy efficiency',
    //     'Environmental health impacts',
    //     'Environmental inequalities',
    //     'Extreme weather',
    //     'Fisheries and aquaculture',
    //     'Forests and forestry',
    //     'Industry',
    //     'Land use',
    //     'Nature protection and restoration',
    //     'Nature-based solutions',
    //     'Noise',
    //     'Plastics',
    //     'Pollution',
    //     'Production and consumption',
    //     'Renewable energy',
    //     'Resource use and materials',
    //     'Road transport',
    //     'Seas and coasts',
    //     'Soil',
    //     'Sustainability challenges',
    //     'Sustainability solutions',
    //     'Sustainable finance',
    //     'Textiles',
    //     'Transport and mobility',
    //     'Urban sustainability',
    //     'Waste and recycling',
    //     'Water',
    //   ],
    // },
    // {
    //   id: 'countries',
    //   title: 'Countries',
    //   facetField: 'spatial',
    //   filterType: 'any:exact',
    //   sortOn: 'alpha',
    //   maxPerSection: 100,
    //   whitelist: [
    //     'Austria',
    //     'Belgium',
    //     'Bulgaria',
    //     'Croatia',
    //     'Cyprus',
    //     'Czechia',
    //     'Denmark',
    //     'Estonia',
    //     'Finland',
    //     'France',
    //     'Germany',
    //     'Greece',
    //     'Hungary',
    //     'Iceland',
    //     'Ireland',
    //     'Italy',
    //     'Latvia',
    //     'Liechtenstein',
    //     'Lithuania',
    //     'Luxembourg',
    //     'Malta',
    //     'Netherlands',
    //     'Norway',
    //     'Poland',
    //     'Portugal',
    //     'Romania',
    //     'Slovakia',
    //     'Slovenia',
    //     'Spain',
    //     'Sweden',
    //     'Switzerland',
    //     'TÃ¼rkiye',
    //     'Albania',
    //     'Bosnia and Herzegovina',
    //     'Kosovo',
    //     'Montenegro',
    //     'North Macedonia',
    //     'Serbia',
    //   ],
    //   icon: {
    //     family: 'CountryFlags',
    //     className: 'facet-option-icon',
    //   },
    // },
    // {
    //   id: 'language',
    //   title: 'Languages',
    //   facetField: 'language',
    //   sortOn: 'custom',
    //   sortOrder: 'asc',
    // },
    // {
    //   id: 'website',
    //   title: 'Websites',
    //   facetField: 'cluster_name',
    //   sortOn: 'count',
    //   sortOrder: 'desc',
    //   icon: {
    //     family: 'Sources',
    //     className: 'facet-option-icon',
    //   },
    // },
  ];

  if (typeof window !== 'undefined') {
    config.searchui.ccaSearch.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }
  // console.log(config.searchui.ccaSearch);

  return config;
}
