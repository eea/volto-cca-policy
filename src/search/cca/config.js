import { mergeConfig } from '@eeacms/search';
import { getClientProxyAddress } from '../utils';
import { vocab } from '../vocabulary';

import facets from './facets';
import views from './views';
import { download_fields } from '../../constants';

const ccaConfig = {
  title: 'ClimateAdapt Main',
  ...views,
};

export const clusters = {
  name: 'op_cluster',
  // field: 'objectProvides',
  clusters: [
    {
      name: 'Climate-ADAPT',
      icon: { name: 'file text' },
      values: ['climate'],
    },
    {
      name: 'Health Observatory',
      icon: { name: 'compass' },
      values: ['observatory'],
    },
    {
      name: 'Mission Portal',
      icon: { name: 'compass' },
      values: ['mission'],
    },
  ].map((cluster) => ({
    ...cluster,
    defaultResultView: 'ClusterHorizontalCardItem',
  })),
};

const cca_build_runtime_mappings = {};
cca_build_runtime_mappings['op_cluster'] = {
  type: 'keyword',
  script: {
    source:
      "def objectProvides = ['Adaptation option','Case study','Guidance','Video','Indicator','Information portal','Organisation','Publication reference','Research and knowledge project','Video','Tool'];def vals = doc['objectProvides'];for (defVal in objectProvides) {for (objVal in vals) {if (objVal.contains(defVal)) {emit('Climate-ADAPT')}}}emit('_all_');if (doc['cca_include_in_search_observatory'][0]) {for (defVal in objectProvides) {for (objVal in vals) {if (objVal.contains(defVal)){emit('Health Observatory')}}}}  if (doc['cca_include_in_mission.keyword'][0].toString() == 'true') {emit('Mission Portal')} else {if (doc['id.keyword'].length==1) { if(doc['id.keyword'][0].toString().contains('climate-adapt.eea.europa.eu/en/mission')) {emit('Mission Portal')}}}",
  },
};

export default function installMainSearch(config) {
  const envConfig = ccaConfig;
  const pjson = require('@eeacms/volto-cca-policy/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.ccaSearch = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/globalsearch',
    index_name: 'data_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
    runtime_mappings: cca_build_runtime_mappings,
    vocab,
  };

  config.searchui.ccaSearch.download_fields = download_fields;

  const { ccaSearch } = config.searchui;

  ccaSearch.contentSectionsParams = {
    enable: true,
    sectionFacetsField: 'op_cluster',
    sections: clusters.clusters,
    clusterMapping: Object.assign(
      {},
      ...clusters.clusters.map(({ name, values }) =>
        Object.assign({}, ...values.map((v) => ({ [v]: name }))),
      ),
    ),
  };

  ccaSearch.permanentFilters.push({
    term: {
      cluster_name: 'cca',
    },
  });

  ccaSearch.facets = facets;

  ccaSearch.initialView.tilesLandingPageParams.sections = [
    {
      id: 'types',
      title: 'Types',
      facetField: 'objectProvides',
      sortOn: 'alpha',
      icon: {
        family: 'Content types',
      },
    },
    {
      id: 'cca_geographic_countries',
      title: 'Countries',
      facetField: 'cca_geographic_countries.keyword',
      // filterType: 'any:exact',
      sortOn: 'alpha',
      // maxPerSection: 100,
      whitelist: [
        'Austria',
        'Belgium',
        'Bulgaria',
        'Croatia',
        'Cyprus',
        'Czechia',
        'Denmark',
        'Estonia',
        'Finland',
        'France',
        'Germany',
        'Greece',
        'Hungary',
        'Iceland',
        'Ireland',
        'Italy',
        'Latvia',
        'Liechtenstein',
        'Lithuania',
        'Luxembourg',
        'Malta',
        'Netherlands',
        'Norway',
        'Poland',
        'Portugal',
        'Romania',
        'Slovakia',
        'Slovenia',
        'Spain',
        'Sweden',
        'Switzerland',
        'TÃ¼rkiye',
        'Albania',
        'Bosnia and Herzegovina',
        'Kosovo',
        'Montenegro',
        'North Macedonia',
        'Serbia',
      ],
      icon: {
        family: 'CountryFlags',
        className: 'facet-option-icon',
      },
    },
  ];

  if (typeof window !== 'undefined') {
    config.searchui.ccaSearch.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }
  let fieldsMatchAll = [
    'cca_climate_impacts.keyword',
    'cca_adaptation_sectors.keyword',
    'cca_adaptation_elements.keyword',
    'cca_key_type_measure.keyword',
  ];
  config.searchui.ccaSearch.facets.forEach((facet) => {
    if (fieldsMatchAll.includes(facet.field)) {
      facet.filterType = 'all';
    }
  });

  return config;
}
