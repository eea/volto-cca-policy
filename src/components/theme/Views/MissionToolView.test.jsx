import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import MissionToolView from './MissionToolView';
import config from '@plone/volto/registry';

config.blocks = {
  blocksConfig: {
    title: {
      view: () => <div>Title Block Component</div>,
    },
  },
};

const mockStore = configureStore();

describe('MissionToolView', () => {
  it('should render the component', () => {
    const data = {
      content: {
        contributors: [],
        description: 'Summary',
        objective: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        short_description: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        free_keywords: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        readiness_for_use: [
          {
            title: 'Tool tested in several case studies',
            token: 'Tool tested in several case studies',
          },
          {
            title: 'Tool broadly used',
            token: 'Tool broadly used',
          },
        ],
        application: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        strengths_weaknesses: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        input: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        output: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        replicability: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        materials: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        website: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        contact: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        associated_project: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        rast_steps: [
          {
            title: 'Step 1. Preparing the ground for adaptation',
            token: 'STEP_1',
          },
          {
            title: 'Step 4. Assessing and selecting adaptation options',
            token: 'STEP_4',
          },
        ],
        geographical_scale: [
          {
            title: 'Item 1',
            token: 'Item 1',
          },
          {
            title: 'Item 2',
            token: 'Item 2',
          },
        ],
        geographical_area: {
          'content-type': 'text/html',
          data: '<p></p>',
          encoding: 'utf8',
        },
        sectors: [
          {
            title: 'Biodiversity',
            token: 'BIODIVERSITY',
          },
          {
            title: 'Coastal areas',
            token: 'COASTAL',
          },
        ],
        title: 'Title here',
      },
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <MissionToolView {...data} />
        </MemoryRouter>
      </Provider>,
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
