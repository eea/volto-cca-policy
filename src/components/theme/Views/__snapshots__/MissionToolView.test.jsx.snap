// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionToolView should render the component 1`] = `
<div
  className="mission-item-view"
>
  <div>
    Title Block Component
  </div>
  <div
    className="ui container"
  >
    <div
      className="ui twelve column grid"
    >
      <div
        className="row"
      >
        <div
          className="eight wide computer twelve wide mobile twelve wide tablet column col-left"
        >
          <h2>
            Objective(s)
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Short description
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Free keywords
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Readiness for use
          </h2>
          <p>
            Tool tested in several case studies, Tool broadly used
          </p>
          <h2>
            Strengths and weaknesses, comparative added value to other similar tools
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Input(s)
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h3>
            Output(s)
          </h3>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Replicability: Cost/effort for (new) usage
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Materials or other support available
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Website and maintenance
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Contact
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
          <h2>
            Associated project(s)
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p></p>",
              }
            }
          />
        </div>
        <div
          className="four wide computer twelve wide mobile twelve wide tablet column col-right"
        >
          <div
            className="ui segment metadata"
          >
            <h5>
              RAST step(s) of relevance:
            </h5>
            <div
              className="ui list"
              role="list"
            >
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                Step 1. Preparing the ground for adaptation
              </div>
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                Step 4. Assessing and selecting adaptation options
              </div>
            </div>
            <h5>
              Geographical scale:
            </h5>
            <p>
              Item 1, Item 2
            </p>
            <h5>
              Geographical area:
            </h5>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p></p>",
                }
              }
            />
            <h5>
              Adaptation sectors:
            </h5>
            <p>
              Biodiversity, Coastal areas
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="styled-slate has--style_name--content-box-gray styled"
    >
      <div
        className="disclaimer content-box"
      >
        <div
          className="content-box-inner"
        >
          <p>
            <strong>
              Disclaimer
            </strong>
            <br />
            The contents and links to third-party items on this Mission webpage are developed by the MIP4Adapt team led by Ricardo, under contract CINEA/2022/OP/0013/SI2.884597 funded by the European Union and do not necessarily reflect those of the European Union, CINEA, or those of the European Environment Agency (EEA) as host of the Climate-ADAPT Platform. Neither the European Union nor CINEA nor the EEA accepts responsibility or liability arising out of or in connection with the information on these pages.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
